etna/search/views.py

```diff
@@ -398,6 +399,30 @@ def get_api_kwargs(self, form: Form) -> Dict[str, Any]:
             sort_order=form.cleaned_data.get("sort_order"),
         )
 
+    def aggregations_params(self) -> Tuple[str]:
+        """
+        The aggregations params may be specific to a bucket and will be filtered upon.
+        There is a max (8) params that can be passed to the api aggregations search param.
+        Returns a list of aggregation params.
+        """
```

Comment:
Instead of adding a new method here, we should update the `Bucket` class to accept this configuration as an optional argument, and pull it from there

Reply:
Sure this is better approach. Implemented the same. Thanks.
1) In order to make the bucket attribute available before the API call, it was required to [set_current_bucket](https://github.com/nationalarchives/ds-wagtail/blob/6fc416a81f17375204be3dabc9e27eb01de023dc/etna/search/views.py#L233) when group value was given.
2) There is a follow on ticket that addresses the required aggregations params for each bucket [DF-641](https://national-archives.atlassian.net/browse/DF-641)